flict/flictlib/project/reader.py

```diff
@@ -140,23 +146,30 @@ def read_project(self, project_file):
             "packages": flat_packages,
         }
 
+    def _relationship_is_dependency(self, relationshiptype):
+        return relationshiptype in DEPENDENCY_TAGS
+
     def _read_spdx_2_2(self, only_packages=None):
         project_name = self.project['name']
         packages = {}
         for pkg in self.project['packages']:
             elem_id = pkg['SPDXID']
             packages[elem_id] = {
                 'id': elem_id,
-                'name': pkg['name'],
-                'version': pkg['versionInfo'],
-                'license': pkg['licenseConcluded'],
-                'description': pkg['description'],
+                'name': pkg.get('name', ""),
+                'version': pkg.get('versionInfo', ""),
+                'license': pkg.get('licenseConcluded'),
+                'description': pkg.get('description'),
                 'dependencies': [],
             }
 
-        if 'relationships' in self.project:
-            for dep in self.project['relationships']:
-                top_package = dep['relatedSpdxElement']
+        for dep in self.project.get('relationships', []):
+            top_package = dep['relatedSpdxElement']
+            relationshiptype = dep['relationshipType']
+            if not self._relationship_is_dependency(relationshiptype):
+                logging.debug(f"relationship {dep.get('spdxElementId')} ignored since {relationshiptype} is not defined to be a dependency tag.")
+                continue
+            if ":" in dep['spdxElementId']:
```

Comment:
IMO it would be better to look for `DocumentRef-` prefixes - just the `:` might not be enough, but that maybe only because of my limited understanding of the data

Reply:
Using both. For spec, see https://spdx.github.io/spdx-spec/v2-draft/relationships-between-SPDX-elements/